packages/core/src/http-client/http-agent.ts

```diff
@@ -29,10 +29,12 @@ export function getAgentConfig(
   const agentType = destination.proxyConfiguration
     ? AgentType.PROXY
     : AgentType.DEFAULT;
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  const certificateOptions = getCertificateOption(destination);
```

Comment:
I don't see shadow name. Could you please share a bit details?

Reply:
for me I got a lint error that the variable in the if block below is shadowed - I do not understand that either.